core/lib/spree/core/controller_helpers/gem_checking.rb

```diff
@@ -0,0 +1,26 @@
+module Spree
+  module Core
+    module ControllerHelpers
+      module GemChecking
+        extend ActiveSupport::Concern
+
+        def backend_available?
+          @backend_available ||= Gem::Specification.find_all_by_name('spree_backend').any?
```

Comment:
We can also do:
```suggestion
          @backend_available ||= Gem.loaded_specs.keys.include?('spree_backend')
```

But regardless, there is a potential issue here I think - this will return `true` when the Gem itself is present but it's not required anywhere i.e. it's in the `Gemfile` but not loaded into the application (I don't remember the semantics here but I'm pretty sure this is how it behaves). In that case, we say that something is available while it's not.

This shouldn't be an issue for most cases but I wouldn't be surprised if it yields something unexpected for some obscure use case. Probably that's why originally the functions checked whether the engines were loaded. WDYT?

Reply:
At the moment it's done using:
`::Rails::Engine.subclasses.map(&:instance).map { |e| e.class.to_s }.include?('Spree::Backend::Engine')`
The reason for changing it to Gem is becouse we had some issues in the past with this way of checking in spree_auth_devise meaning that the order of gems in Gemfile had to be correct or it wouldn't work, which is not ideal.
I don't think it was ever an issue in spree_spree though, so maybe if spree_auth_devise uses functions provided by spree core it wouldn't be an issue (I could test it, try changing order of gems etc.).
Also I think those functions in spree_auth_devise were copied from here: https://github.com/spree/spree/blob/main/core/lib/spree/core/components.rb so I think we can just use that insted of moving it to controller helper.